pkg/auth/authenticator/password/ldappassword/ldap.go

```diff
@@ -84,17 +85,6 @@ func (a *Authenticator) getIdentity(username, password string) (authapi.UserIden
 		return nil, false, nil
 	}
 
-	// Make the connection and bind to it if a bind DN and password were given
```

Comment:
Almost all of the changes were due to the refactor of `ldapclient.Config` and `ldapclient.Client`. I can revert the two lines that were on top of that to reduce duplicated code if you want, but the rest need to stay.